pkg/controller/storagecluster/reconcile.go

```diff
@@ -406,6 +407,11 @@ func (r *ReconcileStorageCluster) validateStorageDeviceSets(sc *ocsv1.StorageClu
 				return fmt.Errorf("failed to validate StorageDeviceSet %d: no StorageClass specified for metadataPVCTemplate", i)
 			}
 		}
+		if ds.DeviceType != "" {
+			if strings.EqualFold(ds.DeviceType, "SSD") || strings.EqualFold(ds.DeviceType, "HDD") || strings.EqualFold(ds.DeviceType, "NVMe") {
```

Comment:
It looks like you wanted to use a not condition here as per the error msg.

Reply:
yes. I'll change that.